lib/slack/bot.ex

```diff
@@ -87,11 +105,11 @@ defmodule Slack.Bot do
       token: token,
       me: rtm.self,
       team: rtm.team,
-      bots: rtm_list_to_map(rtm.bots),
-      channels: rtm_list_to_map(rtm.channels),
-      groups: rtm_list_to_map(rtm.groups),
-      users: rtm_list_to_map(rtm.users),
-      ims: rtm_list_to_map(rtm.ims)
+      bots: bot_to_map(Bots.info(%{token: token}) |> Map.get("bot")),
+      channels: rtm_list_to_map(Channels.list(%{token: token}) |> Map.get("channels")),
```

Comment:
`rtm_list_to_map/1` is a private func only used here, so I think it's better to change the func to simplify this code

```elixir
# before
rtm_list_to_map(Channels.list(%{token: token}) |> Map.get("channels")),

# after
web_api_to_rtm(token, &Channels.list/1, "channels")
```
```

Note that now we know we want specific key, we don't need to build the full map from the response.

Reply:
Feel free to update my PR and change this but I don't think this is simpler or easier to read.